dpreference/src/main/java/me/dozen/dpreference/PrefAccessor.java

```diff
@@ -1,94 +1,166 @@
 package me.dozen.dpreference;
 
-import android.content.ContentValues;
-import android.content.Context;
-import android.database.Cursor;
-import android.net.Uri;
+import ohos.aafwk.ability.DataAbilityHelper;
+import ohos.aafwk.ability.DataAbilityRemoteException;
+import ohos.app.Context;
+import ohos.data.dataability.DataAbilityPredicates;
+import ohos.data.rdb.ValuesBucket;
+import ohos.data.resultset.ResultSet;
+import ohos.hiviewdfx.HiLog;
+import ohos.hiviewdfx.HiLogLabel;
+import ohos.utils.net.Uri;
+import java.util.Arrays;
 
 /**
  * Created by wangyida on 15/12/18.
  */
 class PrefAccessor {
+    private static final HiLogLabel HILOG_LABEL = new HiLogLabel(0, 0, "prefaccessor");
+
+    private PrefAccessor() {
+        throw new IllegalStateException("PrefAccessor class");
+    }
 
     public static String getString(Context context, String name, String key, String defaultValue) {
-        Uri URI = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_STRING);
+        Uri uri = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_STRING);
         String value = defaultValue;
-        Cursor cursor = context.getContentResolver().query(URI, null, null, null, null);
-        if (cursor != null && cursor.moveToFirst()) {
-            value = cursor.getString(cursor.getColumnIndex(PreferenceProvider.PREF_VALUE));
+        ResultSet cursor = null;
+        DataAbilityPredicates dataAbilityPredicates = new DataAbilityPredicates((java.lang.String) null);
+        dataAbilityPredicates.setWhereArgs(Arrays.asList());
+        dataAbilityPredicates.setOrder(null);
+        try {
+            cursor = DataAbilityHelper.creator(context).query(uri, null, dataAbilityPredicates);
+        } catch (DataAbilityRemoteException e) {
+            HiLog.error(HILOG_LABEL, "exception in getString");
+        }
+        if (cursor != null && cursor.goToFirstRow()) {
+            value = cursor.getString(cursor.getColumnIndexForName(PreferenceProvider.PREF_VALUE));
         }
         IOUtils.closeQuietly(cursor);
         return value;
     }
 
     public static int getInt(Context context, String name, String key, int defaultValue) {
-        Uri URI = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_INT);
+        Uri uri = PreferenceProvider.buildUri(name, key, PreferenceProvider.PREF_INT);
         int value = defaultValue;
-        Cursor cursor = context.getContentResolver().query(URI, null, null, null, null);
-        if (cursor != null && cursor.moveToFirst()) {
-            value = cursor.getInt(cursor.getColumnIndex(PreferenceProvider.PREF_VALUE));
+        DataAbilityPredicates dataAbilityPredicates = new DataAbilityPredicates((java.lang.String) null);
```

Comment:
null values need not be casted. Please check and remove the cast

Reply:
Done